api-gateway/src/main/kotlin/com/saveourtool/save/gateway/security/WebSecurityConfig.kt

```diff
@@ -139,21 +117,7 @@ class WebSecurityConfig(
             // Authenticate by comparing received basic credentials with existing one from DB
             httpBasicSpec.authenticationManager(
                 UserDetailsRepositoryReactiveAuthenticationManager { username ->
-                    // Looking for user in DB by received source and name
-                    require(username.contains("@")) {
```

Comment:
`API` comes to `api-gateway` as basic auth.
`API` doesn't need to add provider (since it supports only basic auth) as the header for source (it's set by Api Gateway).

Api Gateway distinguishes the type of authentication by type of String's token:
- OAuth2Token for oauth
- UsernameAndPassword for basic auth

Reply:
probably, initially we planed somehow to use github tokens in API too, that's why we were need this all
okay

Reply:
previously we connect to backend directly from `API`. Now when we ues K8S, backend is hidden and all communication is from api-gateway.
I think it's the root cause of having `source` in `Authentication` on backend side